site/src/components/Markdown/Markdown.tsx

```diff
@@ -113,7 +113,66 @@ export const Markdown: FC<MarkdownProps> = (props) => {
   );
 };
 
+interface MarkdownProps {
+  /**
+   * The Markdown text to parse and render
+   */
+  children: string;
+
+  className?: string;
+
+  /**
+   * Can override the behavior of the generated elements
+   */
+  components?: Options["components"];
+}
+
+/**
+ * Supports a strict subset of Markdown that bahaves well as inline/confined content.
+ */
+export const InlineMarkdown: FC<MarkdownProps> = (props) => {
+  const { children, className, components = {} } = props;
+
+  return (
+    <ReactMarkdown
+      css={markdownStyles}
+      className={className}
+      allowedElements={["p", "em", "strong", "a", "pre", "code"]}
+      unwrapDisallowed
+      components={{
+        p: ({ children }) => <>{children}</>,
+
+        a: ({ href, target, children }) => (
+          <Link href={href} target={target}>
+            {children}
+          </Link>
+        ),
+
+        code: ({ node, className, children, style, ...props }) => (
+          <code
+            css={(theme) => ({
+              padding: "1px 4px",
+              background: theme.palette.divider,
+              borderRadius: 4,
+              color: theme.palette.text.primary,
+              fontSize: 14,
+            })}
+            {...props}
+          >
+            {children}
+          </code>
+        ),
+
+        ...components,
+      }}
+    >
+      {children}
+    </ReactMarkdown>
+  );
+};
+
 export const MemoizedMarkdown = memo(Markdown);
+export const MemoizedInlineMarkdown = memo(InlineMarkdown);
```

Comment:
Can you go into more detail about what the render bottleneck is for this component? The current implementation will always fail to memoize for `ResourceCard` for this line:
```tsx
<MemoizedInlineMarkdown components={{ p }}>
```
Because the default memoization algorithm uses `Object.is` and calls it a day, but the object passed in via `components` will have a new memory reference on each render

Reply:
My gut feeling is to define a custom memoization algorithm via `memo`'s second argument, but I don't know what exactly the memoization is trying to avoid (I'm guessing parsing?)

Reply:
yeah but `memo` checks input props, not the output of the component right?

Reply:
or have I just been holding this wrong for years (also I was just copying what the original component does)

Reply:
https://react.dev/reference/react/memo#returns

Reply:
You're right that it's just checking the inputs of the props, but the problem is how those inputs are defined. `children` is fine, since it's just a string/primitive, and just works with any form of equality. But `components` has more edge cases:

- No `components` prop at all – this is fine, because then `components` is `undefined`, and primitive values are always strictly equal. The inputs never change across re-renders, and the memoization works
- `components` is an object – this is where things break down. The memory reference for the prop changes every render, and because React defaults to shallow equality checks, the comparison always makes React think that it should re-render

Usually, there's two ways to mitigate that:
1. Memoize the `components` object itself via `useMemo`, and pass that in
2. Give `memo` a custom comparison function (which, in this case, could check almost all the props for strict equality, but would check one level deeper for the `components` prop)

Reply:
wow sorry, I had the flow of this really backwards in my head 💀 I just made the object a constant, it should be fine now